library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -133,24 +145,32 @@ Http::FilterDataStatus PlatformBridgeFilter::onData(Buffer::Instance& data, bool
       internal_buffer->drain(internal_buffer->length());
     }
     iteration_state_ = IterationState::Stopped;
-    break;
-  default:
-    PANIC("unsupported status for platform filters");
-  }
+    return Http::FilterDataStatus::StopIterationNoBuffer;
 
-  // TODO(goaway): Current platform implementations expose immutable data, thus any modification
-  // necessitates a full copy. Add 'modified' bit to determine when we can elide the copy. See also
-  // https://github.com/lyft/envoy-mobile/issues/949 for potential future optimization.
-  if (iteration_state_ == IterationState::Ongoing) {
-    data.drain(data.length());
-    data.addBufferFragment(*Buffer::BridgeFragment::createBridgeFragment(result.data));
+  case kEnvoyFilterDataStatusResumeIteration:
+    if (iteration_state_ != IterationState::Stopped) {
+      PANIC("invalid filter state: ResumeIteration  may only be used when filter iteration is "
+            "stopped");
+    }
+    if (result.extra_headers) {
```

Comment:
Why are extra headers only allowed here?

Reply:
They're only allowed where pending headers might still exist. That means onData or onTrailers (or async onResume) when iteration has been previously stopped.